src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -705,21 +839,18 @@ public override bool Equals(object obj)
         /// </summary>
         public bool Equals(SemanticVersion other)
         {
+            // SymVer 2.0 standard requires to ignore 'BuildLabel' (Build metadata).
             return other != null &&
                    (Major == other.Major) && (Minor == other.Minor) && (Patch == other.Patch) &&
-                   string.Equals(Label, other.Label, StringComparison.Ordinal);
+                   string.Equals(PreLabel, other.PreLabel, StringComparison.Ordinal);
         }
 
         /// <summary>
         /// Override <see cref="object.GetHashCode()"/>
         /// </summary>
         public override int GetHashCode()
         {
-            return Utils.CombineHashCodes(
-                Major.GetHashCode(),
-                Minor.GetHashCode(),
-                Patch.GetHashCode(),
-                Label == null ? 0 : Label.GetHashCode());
+            return versionString.GetHashCode();
```

Comment:
`versionString` could be null when running this method.
Besides, if we use the hashcode of  `versionString` for the SemanticVersion, we would have a hashcode collision, would that be a problem? Why not continue to use `Utils.CombineHashCodes`?

Reply:
null - fixed.

versionString is relatively short string - why you believe that CoreFX implementation of `String.GetHashCode()` so bad that we should make more complex hashcode?

Reply:
What I mean is the hashcode of a SemanticsVersion instance would be the same of the hashcode of a string (the versionString), so there is a collision in terms that 2 different objects have the same hashcode.

Reply:
In the [docs](https://msdn.microsoft.com/en-us/library/system.object.gethashcode%28v=vs.110%29.aspx) I don't found that the hash must be different for different objects of different types. It seems we follow the docs - it will be work well with Dictionary and Hashtable types.

Reply:
Maybe I'm just overthinking it. Let's keep it as is.